modules/network/src/test/java/org/apache/ignite/internal/network/StaticNodeFinderTest.java

```diff
@@ -0,0 +1,172 @@
+
+        assertThat(nodes, hasSize(1));
+        InetAddress inetAddress = InetAddress.getByName(nodes.get(0).host());
+        assertTrue(inetAddress.isLoopbackAddress(), nodes.get(0).host() + " must be local");
+    }
+
+    @Test
+    void resolvesNames() throws Exception {
+        Path hostsFilePath = workDir.resolve("hosts");
+        Files.writeString(
+                hostsFilePath,
+                hostsFileContent(Map.of(
+                        "1.2.3.4", "abc.def",
+                        "1.2.3.5", "abc.def",
+                        "4.3.2.1", "def.abc"
+                ))
+        );
+
+        List<NetworkAddress> foundAddresses = findAddressesWithOverriddenNameResolving(
+                List.of(new NetworkAddress("abc.def", 3001), new NetworkAddress("def.abc", 3002)),
+                hostsFilePath
+        );
+
+        assertThat(
+                foundAddresses,
+                containsInAnyOrder(
+                        new NetworkAddress("1.2.3.4", 3001),
+                        new NetworkAddress("1.2.3.5", 3001),
+                        new NetworkAddress("4.3.2.1", 3002)
+                )
+        );
+    }
+
+    @Test
+    void ignoresUnresolvableName() throws Exception {
+        Path hostsFilePath = workDir.resolve("hosts");
+        Files.writeString(hostsFilePath, hostsFileContent(Map.of("1.2.3.4", "abc.def")));
+
+        List<NetworkAddress> foundAddresses = findAddressesWithOverriddenNameResolving(
+                List.of(new NetworkAddress("abc.def", 3001), new NetworkAddress("def.abc", 3002)),
+                hostsFilePath
+        );
+
+        assertThat(foundAddresses, contains(new NetworkAddress("1.2.3.4", 3001)));
+    }
+
+    /**
+     * Invokes {@link StaticNodeFinder} with name resolving overridden with our name->ips mapping.
+     * Does this by starting {@link StaticNodeFinderMain} in another JVM to allow jdk.net.hosts.file property
+     * override name resolving (this defines path to a file in /etc/hosts format).
+     *
+     * @param addresses Addresses to initialize a {@link StaticNodeFinder} with.
+     * @param hostsFilePath Path to hosts file (this file defines the overrides).
+     * @return Found addresses.
+     * @throws IOException If something goes wrong.
+     * @throws InterruptedException If interrupted while waiting for another JVM to terminate.
+     */
+    private static List<NetworkAddress> findAddressesWithOverriddenNameResolving(
+            List<NetworkAddress> addresses,
+            Path hostsFilePath
+    ) throws IOException, InterruptedException {
+        String javaBinaryPath = ProcessHandle.current().info().command().orElseThrow();
+        String javaClassPath = System.getProperty("java.class.path");
+
+        //noinspection UseOfProcessBuilder
+        ProcessBuilder processBuilder = new ProcessBuilder(
+                javaBinaryPath,
+                "-cp", javaClassPath,
+                "-Djdk.net.hosts.file=" + hostsFilePath,
```

Comment:
`System.setProperty` doesn't work for this?

Reply:
This property has effect first time a name resolver is created per JVM, so it's too late to override it at runtime.